etcdserver/api/v3rpc/util.go

```diff
@@ -49,7 +49,7 @@ func togRPCError(err error) error {
 	case etcdserver.ErrTooManyRequests:
 		return rpctypes.ErrTooManyRequests
 
-	case etcdserver.ErrNoLeader:
+	case etcdserver.ErrNoLeader, rpctypes.ErrNoLeader: // ErrNoLeader from leasehttp
```

Comment:
The api layer already calls out to a package called `leasehttp`, so it already has some idea of the lease implementation. I think it's OK for the API layer to know the http request may return an EOF instead of having the leasehttp package make claims about etcd leadership.